test/postgresql.initialization.test.js

```diff
@@ -0,0 +1,43 @@
+// Copyright IBM Corp. 2015. All Rights Reserved.
+// Node module: loopback-connector-postgresql
+// This file is licensed under the Artistic License 2.0.
+// License text available at https://opensource.org/licenses/Artistic-2.0
+
+'use strict';
+
+var should = require('should');
+var DataSource = require('loopback-datasource-juggler').DataSource;
+var postgresql = require('../lib/postgresql.js');
+
+describe('initialization', function() {
+  it('should take pg-pool setting', function(done) {
+    // settings can be empty because we do not actually need to connect
+    var settings = {};
+    var dataSource = new DataSource(require('../'), settings);
+    dataSource.settings.max = 999; // non-default value
+    postgresql.initialize(dataSource, function() {
+      // dataSource is mutated
+      var pool = dataSource.connector.pg.pool;
+      pool._factory.should.have.property('max');
+      pool._factory.max.should.equal(999);
+      done();
+    });
+  }),
+  it('should take url setting and pg-pool setting', function(done) {
+    // settings can be empty because we do not actually need to connect
+    var settings = {url: 'postgres://'};
+    var dataSource = new DataSource(require('../'), settings);
+    dataSource.settings.max = 999; // non-default value
```

Comment:
I also think we should verify the state of the datasource before you run the tests:

Something like:

```
should.not.exist(datasource.connector.clientConfig) // or whatever the defaults are